src/components/CryptoLogic.tsx/CryptoLogic.tsx.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+import {
+  ConteinerData,
+  Title,
+  InputText,
+  ButtonAdd,
+  ButtonText,
+} from './styles';
+import { theme } from '../../utils/theme';
+import { addCrypto } from '../../store/actions';
+import { useAppSelector, useAppDispatch } from '../../Hooks/hooksState';
+import { RootState } from '../../store/index';
+import { Crypto } from '../../interface';
+
+const CryptoLogic = ({ navigation }): JSX.Element => {
+  const [focus, setFocus] = useState(false);
+  const [search, setSearch] = useState('');
+
+  const dispacht = useAppDispatch();
+  const cryptoCollection: Crypto[] = useAppSelector(
+    (state: RootState) => state.cryptos.cryptos
+  );
+  const handleAddCrypto = (): void => {
+    dispacht(addCrypto(search, cryptoCollection));
+    setSearch('');
+    navigation.navigate('Home');
+  };
+  const handleFocus = () => {
+    setFocus(!focus);
```

Comment:
Use the previous value because setState is asynchronous 

setFocus((prev) => !prev);